### PR TITLE
Extend shapes in a caterva array

### PR DESCRIPTION
**Pull request by martaiborra**

### CODE REVIEW

File: caterva/caterva.c [hunk 1]
```diff
@@ -552,7 +552,8 @@ int caterva_blosc_slice(caterva_ctx_t *ctx, void *buffer,
                     CATERVA_ERROR(CATERVA_ERR_BLOSC_FAILED);
                 }
             } else {
-                // memset(data, 0, data_nbytes);
+                // Avoid writing non zero padding from latter chunk
```
Comment (FrancescAlted) [inline caterva/caterva.c]: Maybe latter -> previous ?

File: caterva/caterva.c [hunk 2]
```diff
@@ -1170,3 +1171,124 @@ int caterva_meta_update(caterva_ctx_t *ctx, caterva_array_t *array,
...
+    CATERVA_ERROR(caterva_update_shape(array, ndim, new_shape, array->chunkshape, array->blockshape));
+
+    int64_t nchunks = array->extnitems / array->chunknitems;
+    if (nchunks != old_nchunks) {
+        blosc2_cparams *cparams;
+        blosc2_schunk_get_cparams(array->sc, &cparams);
+        void* chunk;
+        int csize;
+        int nchunks_;
+        for(int i = old_nchunks; i < nchunks; ++i) {
```
Comment (FrancescAlted) [inline caterva/caterva.c]: Space after `for`

File: caterva/caterva.c [hunk 3]
```diff
@@ -1170,3 +1171,124 @@ int caterva_meta_update(caterva_ctx_t *ctx, caterva_array_t *array,
...
+                old = false;
+                break;
+            }
+        }
+        if (old) {
+            index_multidim_to_unidim(nchunk_ndim, ndim, chunks_in_array_strides_old, &new_offsets[i]);
+        }
+    }
+
+    // Copy offsets to int
```
Comment (FrancescAlted) [inline caterva/caterva.c]: Suggested comment:
// Cast offsets back to int

File: caterva/caterva.h [hunk 4]
```diff
@@ -626,4 +626,15 @@ int caterva_print_meta(caterva_array_t *array);
 int caterva_meta_update(caterva_ctx_t *ctx, caterva_array_t *array,
                           caterva_metalayer_t *meta);
 
+/**
+ * @brief Resize the shape of an array
+ *
+ * @param array The array whose shape wil be changed.
```
Comment (FrancescAlted) [inline caterva/caterva.h]: Suggestion: "The array to be resized."

File: tests/test_resize.c [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    cfg.nthreads = 2;
+    cfg.compcodec = BLOSC_ZSTD;
+    caterva_ctx_new(&cfg, &data->ctx);
+
+    // Add parametrizations
+    CUTEST_PARAMETRIZE(itemsize, uint8_t, CUTEST_DATA(
+          1,
+           2,
+           4,
+                                              8,
```
Comment (FrancescAlted) [inline tests/test_resize.c]: Fix indentation above.  It is not better to put all of them in the same line?

**Review by FrancescAlted — COMMENTED** [review]
Looks good!  I just have some comments that should addressed before rebasing.

File: caterva/caterva.c (hunk 1 above)
Comment (martaiborra) [reply caterva/caterva.c]: Fixed in eb616f4d05c1bd76843d158b1b643523602329b7

File: caterva/caterva.c (hunk 2 above)
Comment (martaiborra) [reply caterva/caterva.c]: Fixed in eb616f4d05c1bd76843d158b1b643523602329b7

File: caterva/caterva.c (hunk 3 above)
Comment (martaiborra) [reply caterva/caterva.c]: Fixed in eb616f4d05c1bd76843d158b1b643523602329b7

File: caterva/caterva.h (hunk 4 above)
Comment (martaiborra) [reply caterva/caterva.h]: Fixed in eb616f4d05c1bd76843d158b1b643523602329b7

File: tests/test_resize.c (hunk 5 above)
Comment (martaiborra) [reply tests/test_resize.c]: Fixed in eb616f4d05c1bd76843d158b1b643523602329b7

**Pull request merged**